### PR TITLE
Standarize port name handling between igcollect and igmonplugins

### PR DESCRIPTION
**Pull request by TuxPowered42**

### CODE REVIEW

File: igcollect/switch.py [hunk 1]
```diff
@@ -281,42 +292,15 @@ def get_laggs(snmp, model):
...
-    )
-    if g:
-        # Force 10 MXL port
-        return g.group(2).replace('/', '_')
-    if re.match('\Aethernet[0-9]+/[0-9]+\Z', port_name):
-        # Netiron ports
-        return port_name.replace('/', '_')
+    r = PORT_REGEXP[model].match(port_name)
+    if r:
+        g =  r.groupdict()['port']
```
Comment (lamaral) [inline igcollect/switch.py:301]: You can simplify this with `r.group('port')`

File: igcollect/switch.py [hunk 2]
```diff
@@ -281,42 +292,15 @@ def get_laggs(snmp, model):
...
+        if g:
```
Comment (lamaral) [inline igcollect/switch.py:302]: This if doesn't make sense. 
If the group is not matched and you try to fetch it, you get an IndexError exception.

File: igcollect/switch.py [hunk 3]
```diff
@@ -261,7 +272,7 @@ def get_monitored_ports(snmp, model):
     }
     for port_idx, port_state in port_states.items():
         if not laggs or port_idx not in laggs.keys():
-            port_name = standarize_portname(port_names[port_idx])
+            port_name = standarize_portname(port_names[port_idx], model)
```
Comment (lamaral) [inline igcollect/switch.py:275]: Typo: standardize

**Review by lamaral — CHANGES_REQUESTED** [review]

File: igcollect/switch.py [hunk 4]
```diff
@@ -75,6 +75,15 @@
     },
 }
 
+PORT_REGEXP = {
+    'cisco_ios': re.compile('^(?P<port>(Fa|Gi|Tu)[0-9/]+)$'),
+    'cumulus': re.compile('^(?P<port>swp[0-9]+)$'),
+    'extreme': re.compile('(?P<port>^[0-9]:[0-9]+)$'),
```
Comment (TheFRedFox) [inline igcollect/switch.py:81]:
```suggestion
    'extreme': re.compile('^(?P<port>[0-9]:[0-9]+)$'),
```

File: igcollect/switch.py [hunk 5]
```diff
@@ -281,42 +292,15 @@ def get_laggs(snmp, model):
...
-        '\A(TenGigabitEthernet|fortyGigE) ([0-9]+/[0-9]+)\Z',
-        port_name
-    )
-    if g:
-        # Force 10 MXL port
-        return g.group(2).replace('/', '_')
-    if re.match('\Aethernet[0-9]+/[0-9]+\Z', port_name):
-        # Netiron ports
-        return port_name.replace('/', '_')
+    r = PORT_REGEXP[model].match(port_name)
```
Comment (TheFRedFox) [inline igcollect/switch.py:299]: What about a key that doesn't exist? KeyError being thrown is ok?

File: igcollect/switch.py (hunk 1 above)
Comment (TheFRedFox) [reply igcollect/switch.py:301]: It's not just simplifying. It also makes the call None-safe. No KeyError will be raised for non existing keys.

File: igcollect/switch.py (hunk 2 above)
Comment (TheFRedFox) [reply igcollect/switch.py:302]: It does again after changing `r.groupdict()['port']` to `r.group('port')`

**Review by TheFRedFox — APPROVED** [review]
Rather just trivial stuff. So, early approve.

File: igcollect/switch.py (hunk 2 above)
Comment (lamaral) [reply igcollect/switch.py:302]: Right, I misunderstood the docs. After changing this to r.group(), it returns None instead of the Exception.

Comment (TuxPowered42) [reply igcollect/switch.py:302]: Fixed.

File: igcollect/switch.py (hunk 1 above)
Comment (TuxPowered42) [reply igcollect/switch.py:301]: Fixed.

File: igcollect/switch.py (hunk 3 above)
Comment (TuxPowered42) [reply igcollect/switch.py:275]: Fixed.

File: igcollect/switch.py (hunk 4 above)
Comment (TuxPowered42) [reply igcollect/switch.py:81]: Fixed.

File: igcollect/switch.py (hunk 5 above)
Comment (TuxPowered42) [reply igcollect/switch.py:299]: It's fine. We only support particular types of switches and the script crashing for any new switch is an indicator for us that SNMP OIDs for the new type must be researched and added.

File: igcollect/switch.py [hunk 6]
```diff
@@ -281,44 +292,17 @@ def get_laggs(snmp, model):
...
-        return g.group(2).replace('/', '_')
-    if re.match('\Aethernet[0-9]+/[0-9]+\Z', port_name):
-        # Netiron ports
-        return port_name.replace('/', '_')
-
-    return None
+    r = PORT_REGEXP[model].match(port_name)
+    if not r:
+        return None
+    g =  r.group('port')
```
[the remark this answers is not in the document]
Comment (TuxPowered42) [reply igcollect/switch.py:302]: Fixed.

File: igcollect/switch.py [hunk 7]
```diff
@@ -354,6 +338,11 @@ def cpu_stats(prefix, snmp, model):
         #     5 Secs ( 18.74%)    60 Secs ( 17.84%)   300 Secs ( 18.12%)
         m = re.search('60 Secs \( *([0-9]+)[0-9\.]*%\)', cpu_usage)
         cpu_usage = int(m.group(1))
+    elif model == 'cumulus':
+        # The value is percent idle
+        cpu_usage = 100 - cpu_usage
```
[the remark this answers is not in the document]
Comment (TuxPowered42) [reply igcollect/switch.py:343]: No, it depends on switch type.

**Review by lamaral — APPROVED** [review]

**Pull request merged**